src/transactions/PathPaymentTests.cpp

```diff
@@ -2725,40 +2802,9 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
             // clang-format on
         });
 
-        // For o1a:
-        //     wheatValue = 1 * 1 = 1
-        //     sheepValue = 40 * 1 = 40
-        //     !wheatStays
-        //     price.n < price.d
-        //     sheepSend = floor(1 / 2) = 0
-        //     wheatReceive = ceil(0 * 2 / 1) = 0
         for_versions_from(10, *app, [&] {
-            auto actual = std::vector<ClaimOfferAtom>{};
-            market.requireChanges({{o1a.key, OfferState::DELETED},
-                                   {o1b.key, OfferState::DELETED},
-                                   {o2.key, OfferState::DELETED},
-                                   {o3.key, OfferState::DELETED}},
-                                  [&] {
-                                      actual =
-                                          source
-                                              .pay(destination, cur1, 80, cur4,
-                                                   10, {cur1, cur2, cur3, cur4})
-                                              .success()
-                                              .offers;
-                                  });
-            auto expected = std::vector<ClaimOfferAtom>{
-                o1a.exchanged(0, 0), o1b.exchanged(40, 80),
-                o2.exchanged(20, 40), o3.exchanged(10, 20)};
-            REQUIRE(actual == expected);
-            // clang-format off
-            market.requireBalances(
-                {{source, {{xlm, minBalance4 - 2 * txfee}, {cur1, 0}, {cur2, 0}, {cur3, 0}, {cur4, 0}}},
-                 {mm12a, {{xlm, minBalance3 - 4 * txfee}, {cur1, 0}, {cur2, 1}, {cur3, 0}, {cur4, 0}}},
-                 {mm12b, {{xlm, minBalance3 - 3 * txfee}, {cur1, 80}, {cur2, 0}, {cur3, 0}, {cur4, 0}}},
-                 {mm23, {{xlm, minBalance3 - 3 * txfee}, {cur1, 0}, {cur2, 40}, {cur3, 0}, {cur4, 0}}},
-                 {mm34, {{xlm, minBalance3 - 3 * txfee}, {cur1, 0}, {cur2, 0}, {cur3, 20}, {cur4, 0}}},
-                 {destination, {{xlm, minBalance1 - txfee}, {cur1, 0}, {cur2, 0}, {cur3, 0}, {cur4, 10}}}});
-            // clang-format on
+            REQUIRE_THROWS_AS(mm12a.pay(gateway, cur2, 39),
```

Comment:
should those edge cases be tested differently?

Reply:
These edge cases aren't possible anymore, since it is not possible to have an offer with excess liabilities. Let's not test them starting in version 10.